Enable Neon optimizations within libwebrtc's Opus.

mstyura opened the pull request:
#### 5ddd462f59cd4d419092cec493236d7324719254
<pre>
Enable Neon optimizations within libwebrtc&apos;s Opus.
<a href="https://bugs.webkit.org/show_bug.cgi?id=295575">https://bugs.webkit.org/show_bug.cgi?id=295575</a>

Reviewed by Youenn Fablet.

Added missing files and preprocessor definitions.
Mostly mimic of what was done within BUILD.gn.
This is a reland of <a href="https://commits.webkit.org/298080@main">https://commits.webkit.org/298080@main</a>
with fixed exclude filters.

* Source/ThirdParty/libwebrtc/Configurations/opus.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/298886@main">https://commits.webkit.org/298886@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4f1eca9d8b4f6d6e08d27b0048143f1df1447ebe

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116808 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36472 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27045 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122884 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/67396 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37170 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45073 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88700 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/43112 "Too many flaky failures: fast/canvas/offscreen-no-script-context-crash.html, http/tests/resourceLoadStatistics/only-accept-first-party-cookies-with-third-party-cookie-blocking.html, imported/w3c/web-platform-tests/event-timing/contextmenu.html, imported/w3c/web-platform-tests/trusted-types/TrustedTypePolicyFactory-getAttributeType-event-handler-content-attributes.tentative.html, imported/w3c/web-platform-tests/trusted-types/set-event-handlers-content-attributes.tentative.html, js/dom/jit-set-profiling-access-type-only-for-get-by-id-self.html, js/structuredClone/structured-clone-of-CachedString-in-map.html, js/structuredClone/structured-clone-of-ResizableSharedArrayBuffer.html, storage/indexeddb/modern/idbtransaction-objectstore-failures-private.html, streams/readable-stream-lock-after-worker-terminates-crash.html, transforms/2d/scale-composited.html, webgl/gl-clear-preserve-drawing-buffer-bug.html (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119757 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29634 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104782 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69169 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28697 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22887 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/66549 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99013 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23042 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126018 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43707 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32828 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/97368 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44071 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100983 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97166 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24773 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42493 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/20434 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39715 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43593 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43060 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46399 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44765 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mstyura commented:
I did additional verification locally trying to build `libwebrtc` target for any device platform my XCode provide, here is report:
| Target (libwebrtc) | Status |
|--------------------|--------|
|Any iOS Device (arm64) | success |
|Any iOS Simulator Device (arm64, x86_64) | success |
|Any Mac (arm64, x86_64) | success |
|Any tvOS Device (arm64) | success |
|Any tvOS Simulator Device (arm64, x86_64)| success |
|Any visionOS Device (arm64) | success |
|Apple Vision Pro simulator (arm64) | success |
|Any visionOS Simulator Device (arm64, x86_64) | unable to compile libvpx for x86_64 |
|Any watchOS device| unable to compile due failure in "Check For Inappropriate Objective-C Class Names" |
|Any watchOS Simulator Device (arm64, x86_64) | unable to compile yarn due to "duplicate symbol '_main" |

So I'm quite confident that current state of PR will not break internal builds, as the root cause of original problem was identified - the missing `$(inherited)` in the exclude rule `EXCLUDED_SOURCE_FILE_NAMES[arch=x86_64] = $(inherited) arm/*;`. The absence of `$(inherited)` lead to override the exclude from the rule `EXCLUDED_SOURCE_FILE_NAMES[sdk=iphonesimulator*] = *_sse.c *_sse4_1.c x86_silk_map.c;`.

mstyura commented:
It looks like the [mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30834) failed due to flacky test:  [`Found 1 new test failure: fast/block/transparent-outline-with-and-without-border-radius.html`](https://ews-build.s3-us-west-2.amazonaws.com/macOS-Sequoia-Debug-WK2-Tests-EWS/c9cf77cb-30834-clean-tree/results.html)

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ac0593719cebcaea377f80f147da3b85fd6bf3e6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48818

Ahmad-S792 commented:
> > Unable to determine email address for ['mstyura'] from metadata/contributors.json. Skipping sending email.
> 
> Should I add my email into that file?

Yes - you should add yourself into `contributors.json` file in alphabetical order and your email address should match with the bugzilla one. It will help you login on EWS and reset fail builds to rule out any flakiness + it will auto add labels on your PR etc.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b236a263531ba51c7f7993be5c691c61f5f0776b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48818

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/68e51e1c558b0d65f48bdbe84ddc6d843c876459. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48818

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ea1dcb2786212083ce801f09a29ccbc6d4351749. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48818

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8e61ac6b2e8c84fc12b977bcdbc783c0caea80da. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48818

mstyura committed:
Enable Neon optimizations within libwebrtc's Opus.
https://bugs.webkit.org/show_bug.cgi?id=295575

Reviewed by Youenn Fablet.

Added missing files and preprocessor definitions.
Mostly mimic of what was done within BUILD.gn.
This is a reland of https://commits.webkit.org/298080@main
with fixed exclude filters.

* Source/ThirdParty/libwebrtc/Configurations/opus.xcconfig:
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/298886@main
